docs/commands/lint.md

```diff
@@ -2,239 +2,378 @@
-`--skip-preprocessor`    | `array`   | no           | -           | Ignore certain preprocessors. See the [Skip preprocessor or rule section](#skip-preprocessor-or-rule) below
-`--skip-rule`            | `array`   | no           | -           | Ignore certain rules. See the [Skip preprocessor or rule section](#skip-preprocessor-or-rule) below
-`--version`              | `boolean` | no           | -           | Show version number
+```bash Combined
+redocly lint [--max-problems=<n>] [--config=<path>] [--format=<value>]
+```
 
-## Examples
+| Option | Type | Description |
+|---|---|---|
+| entrypoints | array | Lets you [pass the entrypoints](#how-to-use-entrypoints-to-check-all-or-selected-definitions) (locations) of specific definition files in the command so `lint` only checks those files. |
+| --config | string | Lets you [specify a path to a custom config file](#how-to-reference-a-custom-config-file). |
+| --extends | array | [Extends the existing config](#how-to-activate-rule-sets) by invoking rule sets that apply lots of rules at once. Possible values: `minimal`, `recommended`, `all`. Default: `recommended`. |
+| --format | string | [Applies formatting to the output](#how-to-format-linting-results). Possible values: `codeframe`, `stylish`, `json`, `checkstyle`. Default: `codeframe`. |
+| --generate-ignore-file | boolean | [Generates an ignore file](#how-to-ignore-specific-errors-and-warnings) which tells Redocly CLI to ignore certain warnings/errors when validating the definition. |
+| --help | boolean | [Displays a short help menu](#how-to-open-the-help), showing lint options. |
+| --max-problems | integer | [Truncates the output](#how-to-limit-issues-shown-in-the-output) to show only a specified number of errors and warnings. Default: `100`. |
+| --skip-preprocessor | array | [Ignores specific preprocessors](#how-to-skip-a-rule-or-preprocessor-during-linting). |
+| --skip_rule | array | [Ignores specific custom or built-in rules](#how-to-skip-a-rule-or-preprocessor-during-linting). |
+| --version | boolean | Adding `--version` to any command (not just `lint`) will show your current Redocly CLI version. |
 
-### Entrypoints
+## Usage examples
 
-The `lint` command behaves differently depending on how you pass entrypoints to it and whether the [configuration file](#custom-configuration-file) exists.
+### How to use entrypoints to check all or selected definitions
 
-#### Pass entrypoints directly
+##### Method 1: Pass entrypoints directly
+In this case, `lint` will validate the definitions passed in the command and ignore the `apis` section in `redocly.yaml` or any other customized config file you may be using.
 
 ```bash
-openapi lint openapi/openapi.yaml
+redocly lint openapi/openapi.yaml
 ```
 
-In this case, `lint` will validate the definition(s) passed to the command. The configuration file is ignored.
```

Comment:
I didn't check if the configuration file is ignored. If it is, this is very important documentation.

Reply:
Sorry Adam. I don't know what you mean or what action I should take here.

Reply:
We need to check if this sentence is correct. If it is, it needs to be in the docs.

Reply:
I have several api definition files in my project directory and all are listed in my `redocly.yaml` file under the `apis` object. When I run this command, only the specified definition file is linted, all others are ignored. So I have concluded that this statement is correct.

Reply:
I read this in a different way.

If I have a lint configuration, and I run the lint command on an API using the filename instead of the API name then the lint configurations are ignored and it will use the default lint configuration. I haven't tried to verify what the behavior is though.

Reply:
I ran the command first using the API name then the filename, and both validated:

```
C:\Users\jodyw\Filing Cabinet\Git\redocly-openapi-starter\openapi>redocly lint production@v1
No configurations were defined in extends -- using built in recommended configuration by default.

validating ./production.yaml...
./production.yaml: validated in 40ms

Woohoo! Your OpenAPI definition is valid. 🎉
```

```
C:\Users\jodyw\Filing Cabinet\Git\redocly-openapi-starter\openapi>redocly lint production.yaml
No configurations were defined in extends -- using built in recommended configuration by default.

validating production.yaml...
production.yaml: validated in 38ms

Woohoo! Your OpenAPI definition is valid. 🎉
```

Should our statement in fact read:

In this case, `lint` will only validate the API definition(s) passed in the command (you can use either the API name or filename). For example:

`redocly lint core@v1`
` redocly lint openapi.yaml`